src/components/Profile.tsx

```diff
@@ -54,6 +54,30 @@ const Profile: React.FC = () => {
   React.useEffect(() => {
     fetchStats();
   }, [fetchStats]);
+  const toggleDarkMode = async () => {
+    const isDarkMode = window.darkMode.toggle()
+    const themeSource = document.getElementById('theme-source')
+    if (
+      document
+      && themeSource
+      && 'innerHTML' in themeSource
+    ) {
+      themeSource.innerHTML = isDarkMode ? 'Dark' : 'Light'
+    }
```

Comment:
```suggestion
    if (window.darkMode) {
      const isDarkMode = window.darkMode.toggle()
      const themeSource = document.getElementById('theme-source')
      if (
        document
        && themeSource
        && 'innerHTML' in themeSource
      ) {
        themeSource.innerHTML = isDarkMode ? 'Dark' : 'Light'
      }
    }
```

With `darkMode` now optional, you'll have to make sure it exists before calling the functions on it.